src/fswAlgorithms/effectorInterfaces/solarArrayReference/_UnitTest/test_solarArrayReference.py

```diff
@@ -0,0 +1,227 @@
+#   Creation Date:      January 21, 2023
+#
+
+import pytest
+import os, inspect, random
+import numpy as np
+
+filename = inspect.getframeinfo(inspect.currentframe()).filename
+path = os.path.dirname(os.path.abspath(filename))
+bskName = 'Basilisk'
+splitPath = path.split(bskName)
+
+
+
+# Import all of the modules that we are going to be called in this simulation
+from Basilisk.utilities import SimulationBaseClass
+from Basilisk.utilities import unitTestSupport                   # general support file with common unit test functions
+from Basilisk.fswAlgorithms import solarArrayReference           # import the module that is to be tested
+from Basilisk.utilities import macros
+from Basilisk.utilities import RigidBodyKinematics as rbk
+from Basilisk.architecture import messaging                      # import the message definitions
+from Basilisk.architecture import bskLogging
+
+
+def computeRotationAngle(sigma_RN, rS_N, a1_B, a2_B, theta0):
+
+    RN = rbk.MRP2C(sigma_RN)
+    rS_R = np.matmul(RN, rS_N)
+
+    a2_R = []
+    dotP = np.dot(rS_R, a1_B)
+    for n in range(3):
+        a2_R.append(rS_R[n] - dotP * a1_B[n])
+    a2_R = np.array(a2_R)
+    a2_R_norm = np.linalg.norm(a2_R)
+    if a2_R_norm > 1e-6:
+        a2_R = a2_R / a2_R_norm
+        theta = np.arccos(min(max(np.dot(a2_B, a2_R),-1),1))
+        if np.dot(a1_B, np.cross(a2_B, a2_R)) < 0:
+            theta = -theta
+    else:
+        theta = theta0
+
+    return theta
+
+
+# Uncomment this line is this test is to be skipped in the global unit test run, adjust message as needed.
+# @pytest.mark.skipif(conditionstring)
+# Uncomment this line if this test has an expected failure, adjust message as needed.
+# @pytest.mark.xfail(conditionstring)
+# Provide a unique test method name, starting with 'test_'.
+# The following 'parametrize' function decorator provides the parameters and expected results for each
+# of the multiple test runs for this test.  Note that the order in that you add the parametrize method
+# matters for the documentation in that it impacts the order in which the test arguments are shown.
+# The first parametrize arguments are shown last in the pytest argument list
+@pytest.mark.parametrize("rS_N", [[1, 0, 0],
+                                  [0, 0, 1]])
+@pytest.mark.parametrize("sigma_BN", [[0.1, 0.2, 0.3],
+                                      [0.5, 0.4, 0.3]])
+@pytest.mark.parametrize("sigma_RN", [[0.3, 0.2, 0.1],
+                                      [0.9, 0.7, 0.8]])
+@pytest.mark.parametrize("bodyFrame", [0, 1])                                      
+@pytest.mark.parametrize("accuracy", [1e-12])
+
+
+# update "module" in this function name to reflect the module name
+def test_solarArrayRotation(show_plots, rS_N, sigma_BN, sigma_RN, bodyFrame, accuracy):
+    r"""
+    **Validation Test Description**
+
+    This unit test verifies the correctness of the output reference angle computed by the :ref:`solarArrayRotation`.
```

Comment:
typo here in the link name.  :ref:`solarArrayRotation` should be :ref:`solarArrayReference`

Reply:
Fixed it.